k1te-backend/src/main/java/ua/com/pragmasoft/k1te/backend/router/domain/Channels.java

```diff
@@ -15,5 +17,15 @@ public interface Channels {
 
   Member find(String channel, String memberId);
 
+  Integer findPinnedMessage(Member from, Member to);
```

Comment:
Does it return messageId? Channels is a connector-neutral interface, that's why I modelled message Ids are strings not numbers. Some connectors may have String message IDs (like UUIDs), and while it's relatively easy to convert number to string, not all strings can be easily converted to numbers. 

I understand this complicates ordering.. Probably to futureproof the design the only solution I see is to add GSI by timestamp and order by time. Padding strings looks too hacky, especially as UUIDs may not be monotonic (while some are, e.g. https://www.ietf.org/archive/id/draft-peabody-dispatch-new-uuid-format-03.html)

Reply:
Just checked, WhatsApp and RCS both have string message ids.

Reply:
You are right, I didn't think about it in this way. I guess we should try to make use of GSI on timestamp field in Message History, and I will refactor pin/unpin logic to utilize String instead of Integer and just convert it before pin/unpin.